source/migration/BuildTasks/Xamarin.AndroidX.Migration.targets

```diff
@@ -149,48 +149,63 @@
     -->
     <Target Name="_AndroidXCecilfyShrinkFindFiles"
             Condition="'$(_AndroidXShouldRunMigration)' == 'true' and '$(AndroidLinkMode)' != 'none'">
+
+        <ItemGroup>
+            <_AndroidXResolvedUserAssemblies Include="@(ResolvedUserAssemblies)"
+                                             Condition="'%(ResolvedUserAssemblies.AndroidXSkipAndroidXMigration)' != 'true'" />
+        </ItemGroup>
+
+        <!-- pre Xamarin.Android 10.1 uses 'linksrc' -->
         <ItemGroup>
-            <_AndroidXResolvedUserAssemblies Condition="'%(ResolvedUserAssemblies.AndroidXSkipAndroidXMigration)' != 'true'"
-                                         Include="@(ResolvedUserAssemblies)" />
-            <_AndroidXFileToCecilfy Condition="'$(MonoAndroidLinkerInputDir)' != ''"
-                                    Include="@(_AndroidXResolvedUserAssemblies->'$(MonoAndroidLinkerInputDir)%(Filename)%(Extension)')" />
+            <_AndroidXFileToCecilfy Include="@(_AndroidXResolvedUserAssemblies->'$(MonoAndroidLinkerInputDir)%(Filename)%(Extension)')" 
+                                    Condition="'$(MonoAndroidLinkerInputDir)' != ''" />
         </ItemGroup>
+
     </Target>
 
-    <!-- In Xamarin.Android 10.1, linksrc does not exist -->
+    <!-- Xamarin.Android 10.1 uses the originals, so copy to 'androidx\cecil' -->
     <Target Name="_AndroidXCecilfyShrinkCopyFiles"
             Condition="'$(_AndroidXShouldRunMigration)' == 'true' and '$(AndroidLinkMode)' != 'none' and '$(MonoAndroidLinkerInputDir)' == ''"
-            Inputs="@(ResolvedAssemblies)"
-            Outputs="@(ResolvedAssemblies->'$(IntermediateOutputPath)androidx\cecil\%(Filename)%(Extension)')">
+            Inputs="@(_AndroidXResolvedUserAssemblies)"
+            Outputs="@(_AndroidXResolvedUserAssemblies->'$(IntermediateOutputPath)androidx\cecil\%(Filename)%(Extension)')">
+
         <ItemGroup>
-            <_AndroidXResolvedPdbs Include="@(ResolvedAssemblies->'%(RootDir)%(Directory)%(Filename).pdb')" />
-            <_AndroidXResolvedMdbs Include="@(ResolvedAssemblies->'%(RootDir)%(Directory)%(Filename)%(Extension).mdb')" />
+            <_AndroidXResolvedPdbs Include="@(_AndroidXResolvedUserAssemblies->'%(RootDir)%(Directory)%(Filename).pdb')" />
+            <_AndroidXResolvedMdbs Include="@(_AndroidXResolvedUserAssemblies->'%(RootDir)%(Directory)%(Filename)%(Extension).mdb')" />
```

Comment:
No need to copy ALL the assemblies, just the ones we know we are going to Cecil.